sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -1478,18 +1478,54 @@ class DataFrame private[sql](
 
   /**
    * Returns the first `n` rows in the [[DataFrame]].
+   *
+   * Running take requires moving data into the application's driver process, and doing so on a
+   * very large dataset can crash the driver process with OutOfMemoryError.
```

Comment:
and doing so with a very large `n`?  as phrased now it seems more dangerous than it is.